### DISCUSSION TITLE
constについてお教えください。

### DISCUSSION THREAD

**Discussion by YamadaMiz** [asciidwango/js-primer#1740] [質問] [answered]
## 該当ページ

- [ ] 質問があるページのURLを入れてください
- [ ] https://jsprimer.net/basic/variables/

## 質問内容
「またconstには、変数名の命名規則はなく、」のところの意味がよくわかりません．「ここまででconst、let、varでの変数宣言とそれぞれの特徴について見てきました。 どのキーワードにおいても宣言できる変数に利用できる名前のルールは同じです。」とあるので．constにも，他のキーワードで宣言できる変数名と同じ命名規則があるのではと思ったのですが．
ご教示のほどお願いいたします．

**Comment by azu** [comment, marked as answer]
命名規則は全部大文字やCamelCase(`singleValue` や `firstLine` みたいな単語の区切りを大文字ににするルール)といった名前の付け方のルールのことをいっています。
例えば、言語によっては定数は 大文字で書く といった命名規則があったりしますが、JavaScriptにはそのような命名規則はなく、自由な名前をつけられます。

```js
const TEISU = 10; 
// 定数っぽいもの全部大文字にすることはあるけど、それが言語の仕様として決められた命名規則ではない。
// なので、小文字で定数っぽいものを書いても別に問題はない。人によっては読みにくいというのはあるかもしれないが
const teisu = 10;
```

命名規則は名前の付け方のルールのことを言っています

> どのキーワードにおいても宣言できる変数に利用できる名前のルールは同じです

こちらは、変数名に[使える文字の種類](https://jsprimer.net/basic/variables/#variable-name)は同じということを言っています。

「変数に利用できる名前のルール」は、変数名に使える文字の種類のルールのことを言っています

どっちもルールと言ってちょっと紛らわしいですが、「変数名に利用できる文字の種類は同じです」というのが正しいのかもしれません。
(ただ文字の種類だけじゃなくて、変数は数字から開始できないので、一種のルールとなっててちょっとややこしい…)


変数名に使える文字の種類には決まりがありますが、その変数名の付け方には決まったルールはありませんよ という話になっています。
(慣習的に変数名をcamelCaseで書くとかはあるかもしれませんが、それが仕様で決めらたものではないです)

**Comment by YamadaMiz** [comment]
azu様
丁寧なご回答いただき，たいへんありがとうございます．
「変数名の命名規則」と「変数に利用できる名前のルール」の違いをよくわかっていませんでした．
変数名は，当然，「変数に利用できる名前のルール」に従った上で，letなどで宣言される変数と同じく，constで宣言される変数にも，特に命名規則がないので，constは定数でなく，（再代入できない）変数を定義すると理解するのがよいと理解できました．
たいへんありがとうございます．
